grpc/server.go

```diff
@@ -123,12 +124,21 @@ func (sb *serverBuilder) Build(tlsConfig *tls.Config, statsRegistry prometheus.R
 	// This is the names which are allowlisted at the server level, plus the union
 	// of all names which are allowlisted for any individual service.
 	acceptedSANs := make(map[string]struct{})
+	var acceptedSANsSlice []string
 	for _, service := range sb.cfg.Services {
 		for _, name := range service.ClientNames {
 			acceptedSANs[name] = struct{}{}
+			if !slices.Contains(acceptedSANsSlice, name) {
```

Comment:
nit: this check isn't strictly necessary, because the call to `newServiceAuthChecker(sb.cfg)` on line 156 will perform this same deduplication. Up to you whether you want to rely on that or not.

Reply:
I think I'll leave this as-is. We should definitely consider lightly refactoring this at a later time though.